// @alexa-games/sfb-f/dist/importPlugins/storyBlock.js
"use strict";
/*
 * Copyright 2018 Amazon.com, Inc. and its affiliates. All Rights Reserved.
 *
 * SPDX-License-Identifier: LicenseRef-.amazon.com.-AmznSL-1.0
 *
 * Licensed under the Amazon Software License (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 *   http://aws.amazon.com/asl/
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScenePropertyStoryBlock = exports.SceneStoryBlock = exports.StoryBlock = void 0;
const storyFeatureEnumerator_1 = require("./storyFeatureEnumerator");
class StoryBlock {
    constructor(lines, blockName, range) {
        this.lines = lines;
        this.blockName = blockName;
        this.range = range;
    }
    /**
     * Assembles the whole block into a single multi-line string.
     */
    getText() {
        let result = undefined;
        ;
        for (let i = this.range.start.row; i <= this.range.end.row; i++) {
            if (result) {
                result += '\n';
            }
            else {
                result = '';
            }
            result += this.range.cropToRange(this.lines[i], i);
        }
        return result || '';
    }
}
exports.StoryBlock = StoryBlock;
class SceneStoryBlock extends StoryBlock {
    constructor(lines, blockName, range) {
        super(lines, blockName, range);
    }
    getPropertyEnumerator(type) {
        return new storyFeatureEnumerator_1.ScenePropertyEnumerator(this.lines, this.range, type);
    }
}
exports.SceneStoryBlock = SceneStoryBlock;
class ScenePropertyStoryBlock extends StoryBlock {
    constructor(lines, blockName, range) {
        super(lines, blockName, range);
    }
}
exports.ScenePropertyStoryBlock = ScenePropertyStoryBlock;
