// @alexa-games/sfb-f/dist/importPlugins/storyFeatureEnumerator.js
"use strict";
/*
 * Copyright 2018 Amazon.com, Inc. and its affiliates. All Rights Reserved.
 *
 * SPDX-License-Identifier: LicenseRef-.amazon.com.-AmznSL-1.0
 *
 * Licensed under the Amazon Software License (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 *   http://aws.amazon.com/asl/
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScenePropertyEnumerator = exports.SceneEnumerator = void 0;
const storyBlockFinder_1 = require("./storyBlockFinder");
const storyBlock_1 = require("./storyBlock");
class SceneEnumerator {
    constructor(lines, sceneId) {
        this.finder = new storyBlockFinder_1.StoryBlockFinder({
            blockEndType: storyBlockFinder_1.EndType.NextBlock,
            lines: lines,
            blockStartMatch: storyBlockFinder_1.BlockStartExpressions.SceneMatch,
            blockName: sceneId
        });
    }
    /**
     * Returns the next scene. Returns undefined when no more scenes are available.
     */
    getNext() {
        const result = this.finder.getNextBlock();
        if (result) {
            return new storyBlock_1.SceneStoryBlock(result.lines, result.blockName, result.range);
        }
        return undefined;
    }
}
exports.SceneEnumerator = SceneEnumerator;
class ScenePropertyEnumerator {
    constructor(lines, range, type) {
        this.finder = new storyBlockFinder_1.StoryBlockFinder({
            blockEndType: storyBlockFinder_1.EndType.NextBlock,
            lines: lines,
            range,
            blockName: type,
            blockStartMatch: storyBlockFinder_1.BlockStartExpressions.ScenePropMatch
        });
    }
    /**
     * Returns the next scene property. Returns undefined when no more are available.
     */
    getNext() {
        const result = this.finder.getNextBlock();
        if (result) {
            return new storyBlock_1.ScenePropertyStoryBlock(result.lines, result.blockName, result.range);
        }
        return undefined;
    }
}
exports.ScenePropertyEnumerator = ScenePropertyEnumerator;
