// @alexa-games/sfb-f/dist/importPlugins/userInputHelper.js
"use strict";
/*
 * Copyright 2018 Amazon.com, Inc. and its affiliates. All Rights Reserved.
 *
 * SPDX-License-Identifier: LicenseRef-.amazon.com.-AmznSL-1.0
 *
 * Licensed under the Amazon Software License (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 *   http://aws.amazon.com/asl/
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserInputHelper = void 0;
/**
 * Accessor/Helper for UserInput used as an input for the Driver's story processing.
 */
class UserInputHelper {
    constructor(userInput) {
        this.userInput = userInput;
    }
    /**
     * Gets the HandlerInput generated by the ASK SDK for the current request
     */
    getHandlerInput() {
        return this.userInput.handlerInput;
    }
    getInputValue() {
        return this.userInput.value;
    }
    setInputValue(value) {
        this.userInput.value = value;
    }
    getInputSlots() {
        return this.userInput.slots;
    }
    setInputSlots(slots) {
        this.userInput.slots = slots;
    }
    getInputIntent() {
        return this.userInput.intent;
    }
    setInputIntent(intent) {
        this.userInput.intent = intent;
    }
    addInputSlot(slotName, value) {
        if (!this.userInput.slots) {
            this.userInput.slots = [];
        }
        this.userInput.slots.push({
            name: slotName,
            value: value
        });
    }
    clearUserInput() {
        delete this.userInput.intent;
        delete this.userInput.slots;
        delete this.userInput.value;
    }
    getUserInput() {
        return this.userInput;
    }
}
exports.UserInputHelper = UserInputHelper;
