// @alexa-games/sfb-f/dist/driver/driverEntity.js
"use strict";
/*
 * Copyright 2018 Amazon.com, Inc. and its affiliates. All Rights Reserved.
 *
 * SPDX-License-Identifier: LicenseRef-.amazon.com.-AmznSL-1.0
 *
 * Licensed under the Amazon Software License (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 *   http://aws.amazon.com/asl/
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlexaAudioType = exports.AudioBlendOption = exports.BuiltInUserInput = void 0;
exports.BuiltInUserInput = {
    "Resume Story": {
        intent: "LaunchRequest",
    },
    "Exit Story": {
        intent: "ExitRequest"
    }
};
var AudioBlendOption;
(function (AudioBlendOption) {
    AudioBlendOption["TRIM"] = "trim";
    AudioBlendOption["LONGEST"] = "longest";
})(AudioBlendOption = exports.AudioBlendOption || (exports.AudioBlendOption = {}));
var AlexaAudioType;
(function (AlexaAudioType) {
    AlexaAudioType["POLLY"] = "polly";
    AlexaAudioType["AUDIO"] = "audio";
    AlexaAudioType["TEXT"] = "text";
})(AlexaAudioType = exports.AlexaAudioType || (exports.AlexaAudioType = {}));
