// @alexa-games/sfb-f/dist/extensions/coreExtensions/VoiceOverExtension.js
"use strict";
/*
 * Copyright 2018 Amazon.com, Inc. and its affiliates. All Rights Reserved.
 *
 * SPDX-License-Identifier: LicenseRef-.amazon.com.-AmznSL-1.0
 *
 * Licensed under the Amazon Software License (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 *   http://aws.amazon.com/asl/
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoiceOverExtension = void 0;
const _https = __importStar(require("https"));
const url = __importStar(require("url"));
const crypto_1 = require("crypto");
async function urlExists(urlString, http) {
    return new Promise((resolve) => {
        let options = {
            method: 'HEAD',
            host: url.parse(urlString).host,
            port: 443,
            path: url.parse(urlString).pathname
        };
        let request = http.request(options, function (r) {
            resolve(r.statusCode == 200);
        });
        request.end();
    });
}
/**
 * Replaces a section of speech surrounded by voice over marker with appropriate sound file.
 * A section surrounded by voice over tags (<vo> </vo>) is considered the voice over section.
 * Also, prints out the script and the generated file name for the section.
 */
class VoiceOverExtension {
    /**
     * @param urlFormat string showing the url pattern. {{file_name}} is replaced with the auto generated file name.
     */
    constructor(urlFormat, http) {
        this.urlFormat = urlFormat;
        this.recordingScript = "";
        this.http = http || _https;
    }
    async extendSourceContent(sourceHelper) {
    }
    async extendImportedContent(metadataHelper) {
        let voiceOverRegex = /<[\s]*?vo[\s]*?>([\s\S]+?)<[\s]*?\/[\s]*?vo[\s]*?>/g;
        let recordingScript = "";
        let urlFormat = this.urlFormat;
        let nonExistingURLs = [];
        let sceneString = JSON.stringify(metadataHelper.getAllScenes());
        let resultString = JSON.stringify(metadataHelper.getAllScenes());
        let voTagMatch = voiceOverRegex.exec(sceneString);
        let alreadyCaptured = {};
        while (voTagMatch) {
            let match = voTagMatch[0];
            let p1 = voTagMatch[1];
            let scriptHash = crypto_1.createHash('md5').update(p1.trim().toLowerCase().replace(/[\s]+/g, "")).digest('hex');
            let fileName = `${scriptHash}.mp3`;
            let fileURL = urlFormat.replace("{{file_name}}", fileName);
            if (!alreadyCaptured[scriptHash]) {
                alreadyCaptured[scriptHash] = true;
                let scriptContent = p1.replace(/\\n/g, "\n").replace(/\\t/g, "\t")
                    .replace(/\\'/g, "'").replace(/\\"/g, "\"");
                recordingScript += `\n\n[FILENAME] = '${fileName}'`;
                recordingScript += "\n" + scriptContent + "\n";
            }
            let exists = await urlExists(fileURL, this.http);
            if (exists) {
                resultString = resultString.replace(match, `<audio src='${fileURL}' />`);
            }
            else {
                nonExistingURLs.push(fileURL);
                resultString = resultString.replace(match, p1);
            }
            voTagMatch = voiceOverRegex.exec(sceneString);
        }
        this.recordingScript = recordingScript;
        if (nonExistingURLs.length > 0) {
            console.warn(`Voice-over file do not exist for the following URLs: ${JSON.stringify(nonExistingURLs, null, 4)}`);
        }
        metadataHelper.setAllScenes(JSON.parse(resultString));
    }
    getRecordingScript() {
        return this.recordingScript;
    }
}
exports.VoiceOverExtension = VoiceOverExtension;
