// @alexa-games/sfb-f/dist/test/unitTests/extensions/voiceOverExtensionTest.spec.js
"use strict";
/*
 * Copyright 2018 Amazon.com, Inc. and its affiliates. All Rights Reserved.
 *
 * SPDX-License-Identifier: LicenseRef-.amazon.com.-AmznSL-1.0
 *
 * Licensed under the Amazon Software License (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 *   http://aws.amazon.com/asl/
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const VoiceOverExtension_1 = require("./../../../extensions/coreExtensions/VoiceOverExtension");
const storyMetadataHelper_1 = require("../../../importPlugins/storyMetadataHelper");
const storyMetadata_1 = require("./../../../story/storyMetadata");
const sourceContentHelper_1 = require("../../../importPlugins/sourceContentHelper");
const assert_1 = require("assert");
const sinon = __importStar(require("sinon"));
describe("Voice Over Extension Test", function () {
    it("No voice over tag in story.", async function () {
        const storyhelper = new storyMetadataHelper_1.StoryMetadataHelper(TEST_STORY);
        const voiceOverExtension = new VoiceOverExtension_1.VoiceOverExtension('https://my.custom.url/{{filename}}');
        await voiceOverExtension.extendImportedContent(storyhelper);
        const resultingScript = voiceOverExtension.getRecordingScript();
        assert_1.strict.equal(resultingScript, "");
        assert_1.strict.equal(storyhelper.getSceneNarration('start'), TEST_STORY.scenes[0].contents[0].narration);
    });
    it("Tagged narration - file exists.", async function () {
        const mockHTTP = {
            request: (options, callback) => {
                callback({
                    statusCode: 200
                });
                return {
                    end: () => null
                };
            }
        };
        sinon.spy(mockHTTP, "request");
        const storyhelper = new storyMetadataHelper_1.StoryMetadataHelper(VO_STORY);
        const voiceOverExtension = new VoiceOverExtension_1.VoiceOverExtension('https://localhost/{{filename}}', mockHTTP);
        await voiceOverExtension.extendImportedContent(storyhelper);
        const resultingScript = voiceOverExtension.getRecordingScript();
        assert_1.strict.notEqual(resultingScript, "");
        assert_1.strict.notEqual(storyhelper.getSceneNarration('start'), TEST_STORY.scenes[0].contents[0].narration);
    });
    it("Tagged narration - file does not exists.", async function () {
        const mockHTTP = {
            request: (options, callback) => {
                callback({
                    statusCode: 500
                });
                return {
                    end: () => null
                };
            }
        };
        sinon.spy(mockHTTP, "request");
        const storyhelper = new storyMetadataHelper_1.StoryMetadataHelper(VO_STORY);
        const voiceOverExtension = new VoiceOverExtension_1.VoiceOverExtension('https://localhost/{{filename}}', mockHTTP);
        await voiceOverExtension.extendImportedContent(storyhelper);
        const resultingScript = voiceOverExtension.getRecordingScript();
        assert_1.strict.notEqual(resultingScript, "");
        assert_1.strict.equal(storyhelper.getSceneNarration('start'), TEST_STORY.scenes[0].contents[0].narration);
    });
    // set up to check and make sure http correctly detects existing mp3 via the url
    it("Tagged narration - file exists - integration test", async function () {
        const storyhelper = new storyMetadataHelper_1.StoryMetadataHelper(VO_STORY);
        const voiceOverExtension = new VoiceOverExtension_1.VoiceOverExtension('https://sfb-framework.s3.amazonaws.com/examples/sounds/Copy_Machine_1.mp3');
        await voiceOverExtension.extendImportedContent(storyhelper);
        const resultingScript = voiceOverExtension.getRecordingScript();
        assert_1.strict.notEqual(resultingScript, "");
        assert_1.strict.equal(storyhelper.getSceneNarration('start'), "<audio src=\'https://sfb-framework.s3.amazonaws.com/examples/sounds/Copy_Machine_1.mp3\' />");
    });
    it("Unused extension method 'extendSourceContent' does not throw when called", async function () {
        const voiceOverExtension = new VoiceOverExtension_1.VoiceOverExtension('https://sfb-framework.s3.amazonaws.com/examples/sounds/Copy_Machine_1.mp3');
        await voiceOverExtension.extendSourceContent(new sourceContentHelper_1.SourceContentHelper([
            {
                id: "test.abc",
                text: "@start"
            }
        ]));
    });
});
const TEST_STORY = {
    pluginName: "something",
    scenes: [
        {
            contents: [
                {
                    narration: "testing narration",
                    sceneDirections: [
                        {
                            directionType: storyMetadata_1.InstructionType.REPROMPT,
                            parameters: {
                                message: "this is a reprompt"
                            }
                        },
                        {
                            directionType: storyMetadata_1.InstructionType.REPROMPT,
                            parameters: {
                                message: "this is a recap"
                            }
                        },
                        {
                            directionType: storyMetadata_1.InstructionType.CHOICE,
                            parameters: {
                                "utterances": [
                                    "restart",
                                    "start over"
                                ],
                                "saveToHistory": "true",
                                "directions": [
                                    {
                                        "directionType": "bookmark",
                                        "parameters": {
                                            "variableName": "bookmark"
                                        },
                                    },
                                    {
                                        "directionType": "go to",
                                        "parameters": {
                                            "target": "restart confirm",
                                            "targetSceneProperty": "narration"
                                        },
                                    }
                                ]
                            }
                        }
                    ]
                },
            ],
            id: "start"
        }
    ],
    storyID: "something",
    storyTitle: "something"
};
const VO_STORY = {
    pluginName: "something",
    scenes: [
        {
            contents: [
                {
                    narration: "<vo>testing narration</vo>",
                    sceneDirections: [
                        {
                            directionType: storyMetadata_1.InstructionType.REPROMPT,
                            parameters: {
                                message: "this is a reprompt"
                            }
                        },
                        {
                            directionType: storyMetadata_1.InstructionType.REPROMPT,
                            parameters: {
                                message: "this is a recap"
                            }
                        },
                        {
                            directionType: storyMetadata_1.InstructionType.CHOICE,
                            parameters: {
                                "utterances": [
                                    "restart",
                                    "start over"
                                ],
                                "saveToHistory": "true",
                                "directions": [
                                    {
                                        "directionType": "bookmark",
                                        "parameters": {
                                            "variableName": "bookmark"
                                        },
                                    },
                                    {
                                        "directionType": "go to",
                                        "parameters": {
                                            "target": "restart confirm",
                                            "targetSceneProperty": "narration"
                                        },
                                    }
                                ]
                            }
                        }
                    ]
                },
            ],
            id: "start"
        }
    ],
    storyID: "something",
    storyTitle: "something"
};
