// @alexa-games/sfb-f/dist/story/storyMetadata.js
"use strict";
/*
 * Copyright 2018 Amazon.com, Inc. and its affiliates. All Rights Reserved.
 *
 * SPDX-License-Identifier: LicenseRef-.amazon.com.-AmznSL-1.0
 *
 * Licensed under the Amazon Software License (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 *   http://aws.amazon.com/asl/
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstructionType = exports.BuiltInScenes = void 0;
var BuiltInScenes;
(function (BuiltInScenes) {
    BuiltInScenes["ResumeScene"] = "Resume";
    BuiltInScenes["PauseScene"] = "Pause";
    BuiltInScenes["StartScene"] = "Start";
})(BuiltInScenes = exports.BuiltInScenes || (exports.BuiltInScenes = {}));
var InstructionType;
(function (InstructionType) {
    InstructionType["CONDITION"] = "condition";
    InstructionType["CHOICE"] = "choice";
    InstructionType["BGM"] = "backgroundmusic";
    InstructionType["INCREASE"] = "increase";
    InstructionType["REDUCE"] = "reduce";
    InstructionType["MULTIPLY"] = "multiply";
    InstructionType["DIVIDE"] = "divide";
    InstructionType["MODULUS"] = "modulus";
    InstructionType["SET"] = "set";
    InstructionType["FLAG"] = "flag";
    InstructionType["UNFLAG"] = "unflag";
    InstructionType["CLEAR"] = "clear";
    InstructionType["RECAP"] = "recap";
    InstructionType["ROLL"] = "roll dice";
    InstructionType["ADD_TO_INVENTORY"] = "add inventory";
    InstructionType["ADD_ITEM"] = "add item";
    InstructionType["REMOVE_ITEM"] = "remove item";
    InstructionType["REMOVE_FIRST"] = "remove first";
    InstructionType["REMOVE_LAST"] = "remove last";
    InstructionType["ADD_COLLECTION"] = "add collection";
    InstructionType["CUSTOM"] = "custom";
    InstructionType["REPROMPT"] = "reprompt";
    InstructionType["GET_TIME"] = "get time";
    InstructionType["VISUALS"] = "visuals";
    InstructionType["SLOT"] = "slot";
    InstructionType["BOOKMARK"] = "bookmark";
    InstructionType["GO_TO"] = "go to";
    InstructionType["SAVE_AND_GO"] = "save and go";
    InstructionType["END"] = "ending";
    InstructionType["RESTART"] = "restart";
    InstructionType["RESUME"] = "resume";
    InstructionType["RETURN"] = "return";
    InstructionType["REPEAT"] = "repeat";
    InstructionType["BACK"] = "back";
    InstructionType["PAUSE"] = "pause";
    InstructionType["REPEAT_REPROMPT"] = "repeat reprompt";
})(InstructionType = exports.InstructionType || (exports.InstructionType = {}));
