// @alexa-games/sfb-f/dist/extensions/SFBExtension.js
"use strict";
/*
 * Copyright 2018 Amazon.com, Inc. and its affiliates. All Rights Reserved.
 *
 * SPDX-License-Identifier: LicenseRef-.amazon.com.-AmznSL-1.0
 *
 * Licensed under the Amazon Software License (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 *   http://aws.amazon.com/asl/
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.isImporterExtension = exports.isDriverExtension = exports.isInstructionExtension = void 0;
function isInstructionExtension(extension) {
    return true; // Since "InstructionExtension" doesn't have required methods, any object is "technically" an InstructionExtension
}
exports.isInstructionExtension = isInstructionExtension;
function isDriverExtension(extension) {
    return extension.pre !== undefined;
}
exports.isDriverExtension = isDriverExtension;
function isImporterExtension(extension) {
    return extension.extendImportedContent !== undefined;
}
exports.isImporterExtension = isImporterExtension;
var GlobalDirectionsExtension_1 = require("./coreExtensions/GlobalDirectionsExtension");
Object.defineProperty(exports, "GlobalDirectionsExtension", { enumerable: true, get: function () { return GlobalDirectionsExtension_1.GlobalDirectionsExtension; } });
var LocalizationExtension_1 = require("./coreExtensions/LocalizationExtension");
Object.defineProperty(exports, "LocalizationExtension", { enumerable: true, get: function () { return LocalizationExtension_1.LocalizationExtension; } });
var SnippetExtension_1 = require("./coreExtensions/SnippetExtension");
Object.defineProperty(exports, "SnippetExtension", { enumerable: true, get: function () { return SnippetExtension_1.SnippetExtension; } });
var VoiceOverExtension_1 = require("./coreExtensions/VoiceOverExtension");
Object.defineProperty(exports, "VoiceOverExtension", { enumerable: true, get: function () { return VoiceOverExtension_1.VoiceOverExtension; } });
