// @alexa-games/sfb-polly/dist/index.js
"use strict";
/*
 * Copyright 2018 Amazon.com, Inc. and its affiliates. All Rights Reserved.
 *
 * SPDX-License-Identifier: LicenseRef-.amazon.com.-AmznSL-1.0
 *
 * Licensed under the Amazon Software License (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 *   http://aws.amazon.com/asl/
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var audioMixer_1 = require("./audioMixer");
Object.defineProperty(exports, "AudioMixer", { enumerable: true, get: function () { return audioMixer_1.AudioMixer; } });
Object.defineProperty(exports, "TrimOption", { enumerable: true, get: function () { return audioMixer_1.TrimOption; } });
Object.defineProperty(exports, "SequenceType", { enumerable: true, get: function () { return audioMixer_1.SequenceType; } });
Object.defineProperty(exports, "MixMode", { enumerable: true, get: function () { return audioMixer_1.MixMode; } });
var s3AudioAccessor_1 = require("./audioAccessor/s3AudioAccessor");
Object.defineProperty(exports, "S3AudioAccessor", { enumerable: true, get: function () { return s3AudioAccessor_1.S3AudioAccessor; } });
var pollyUtil_1 = require("./pollyUtil");
Object.defineProperty(exports, "PollyUtil", { enumerable: true, get: function () { return pollyUtil_1.PollyUtil; } });
Object.defineProperty(exports, "PollyOutFormat", { enumerable: true, get: function () { return pollyUtil_1.PollyOutFormat; } });
